dist/update-release-notes.go

```diff
@@ -0,0 +1,449 @@
+	releaseMetaData := releaseMetaData{
+		TagVersionReleaseBodyMap: make(map[string]string),
+		ReleaseVersionTagMap:     make(map[string]string),
+		ReleaseVersionTitleMap:   make(map[string]string),
+		TagReleaseVersionMap:     make(map[string]string),
+		LatestReleaseVersion:     "",
+	}
+
+	for i, release := range releaseTagValidationInput.changeLog.versions {
+		releaseVersionHasTag := false
+		for _, tagVersion := range releaseTagValidationInput.tags {
+			if release.number == tagVersion.Name {
+				releaseVersionHasTag = true
+			}
+		}
+
+		if releaseVersionHasTag {
+			if i == 0 {
+				releaseMetaData.LatestReleaseVersion = release.number
+			}
+			releaseMetaData.ReleaseVersionTitleMap[release.number] = release.title
+			releaseMetaData.ReleaseVersionTagMap[release.number] = release.number
+			releaseMetaData.TagVersionReleaseBodyMap[release.number] = releaseTagValidationInput.releaseNotes[i]
+		} else {
+			fmt.Println(redColor+"WARNING: release", release, "missing tag"+resetColor)
+		}
+	}
+
+	for _, tag := range releaseTagValidationInput.tags {
+		tagHasVersionNumber := false
+		for _, release := range releaseTagValidationInput.changeLog.versions {
+			if tag.Name == release.number {
+				tagHasVersionNumber = true
+			}
+		}
+		if tagHasVersionNumber {
+			releaseMetaData.TagReleaseVersionMap[tag.Name] = tag.Name
+		} else {
+			fmt.Println(redColor+"WARNING: tag", tag.Name, "missing changelog entry"+resetColor)
+		}
+	}
+	return releaseMetaData
+}
+
+func splitAndEncodeURLPath(urlPath string) (string, string) {
+	urlPathSplit := strings.Split(urlPath, "/")
+	owner, name := url.QueryEscape(urlPathSplit[0]), url.QueryEscape(urlPathSplit[1])
+	return owner, name
+}
+
+func getReleases(authToken string, repoPath string) []releaseForUpdate {
+	repoOwner, repoName := splitAndEncodeURLPath(repoPath)
+	releasePath := fmt.Sprintf("https://api.github.com/repos/%v/%v/releases", repoOwner, repoName)
+	req, err := http.NewRequest("GET", releasePath, nil)
+	req.Header.Set("Accept", "application/vnd.github.v3+json")
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "token "+authToken)
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		log.Fatal(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var eachRelease []releaseForUpdate
+	err = json.Unmarshal(body, &eachRelease)
+	if err != nil {
+		fmt.Println(redColor + string(body) + resetColor)
+		log.Fatal(err)
+	}
+	return eachRelease
+}
+
+func getTagsFromGitHub(tagsRepoPath string) []tag {
+	// https://docs.github.com/en/rest/reference/repos#list-repository-tags
+	tagsRepoOwner, tagsRepoName := splitAndEncodeURLPath(tagsRepoPath)
+	pathToTags := fmt.Sprintf("https://api.github.com/repos/%v/%v/tags", tagsRepoOwner, tagsRepoName)
+	resp, err := http.Get(pathToTags)
+	if err != nil {
+		log.Fatal(err)
+	}
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
`ioutil.ReadAll` is deprecated. Call `io.ReadAll` instead.
```suggestion
	body, err := io.ReadAll(resp.Body)
```